drizzlepac/hlautils/astrometric_utils.py

```diff
@@ -361,15 +365,45 @@ def build_auto_kernel(imgarr, whtarr, fwhm=3.0, threshold=None, source_box=7,
             kernel[:] = 0.0
 
     if kernel.sum() > 0.0:
-        kernel /= kernel.sum() # Normalize the new kernel to a total flux of 1.0
+        kernel /= kernel.sum()  # Normalize the new kernel to a total flux of 1.0
+        log.info("Computing kernel FWHM...{}".format(kernel.sum()))
+        kernel_fwhm = find_fwhm(kernel, fwhm)
+        log.info("Determined FWHM from sample PSF of {:.2f}".format(kernel_fwhm))
     else:
         # Generate a default kernel using a simple 2D Gaussian
+        kernel_fwhm = fwhm
         sigma = fwhm * gaussian_fwhm_to_sigma
-        kernel = Gaussian2DKernel(sigma, x_size=source_box, y_size=source_box)
-        kernel.normalize()
-
-    return kernel
-
+        k = Gaussian2DKernel(sigma, x_size=source_box, y_size=source_box)
+        k.normalize()
+        kernel = k.array
+
+    return kernel, kernel_fwhm
+
+def find_fwhm(psf, default_fwhm):
+    """Determine FWHM for auto-kernel PSF"""
+    daogroup = DAOGroup(crit_separation=8)
+    mmm_bkg = MMMBackground()
+    iraffind = DAOStarFinder(threshold=2.5 * mmm_bkg(psf), fwhm=default_fwhm)
+    fitter = LevMarLSQFitter()
+    sigma_psf = gaussian_fwhm_to_sigma * default_fwhm
+    gaussian_prf = IntegratedGaussianPRF(sigma=sigma_psf)
+    gaussian_prf.sigma.fixed = False
+    itr_phot_obj = IterativelySubtractedPSFPhotometry(finder=iraffind,
+                                                       group_maker=daogroup,
+                                                       bkg_estimator=mmm_bkg,
+                                                       psf_model=gaussian_prf,
+                                                       fitter=fitter,
+                                                       fitshape=(11, 11),
+                                                       niters=2)
+    phot_results = itr_phot_obj(psf)
+
+    psf_row = np.where(phot_results['flux_fit'] == phot_results['flux_fit'].max())[0][0]
+    print(phot_results['x_fit', 'y_fit', 'flux_fit', 'sigma_fit'])
```

Comment:
This print statement could be removed as it is really only useful for debugging.

Reply:
Issue resolved in commit [c631ee9](https://github.com/spacetelescope/drizzlepac/pull/455/commits/c631ee978a71b8519922f8783c215cb65a6165a8)